presto-main/src/main/java/com/facebook/presto/sql/planner/NodePartitioningManager.java

```diff
@@ -127,15 +127,21 @@ public NodePartitionMap getNodePartitioningMap(Session session, PartitioningHand
 
         List<InternalNode> bucketToNode;
         NodeSelectionStrategy nodeSelectionStrategy = connectorBucketNodeMap.getNodeSelectionStrategy();
+        boolean cacheable;
         switch (nodeSelectionStrategy) {
             case HARD_AFFINITY:
+                bucketToNode = getFixedMapping(connectorBucketNodeMap);
+                cacheable = false;
```

Comment:
I might miss something, but why is hard affinity not cacheable?

Reply:
Hard affinity is for connectors that use its own storage like Raptor, we don't want to do cache for them

Reply:
Okay, sounds good. I thought hard affinity could also apply for disaggregated storage (though should never be more preferable than soft affinity), but let's not worry about it for now.